# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

**Chosen Lane:** Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because businesses often manage hundreds or even thousands of webpages, making it difficult to decide which pages should be reviewed first. Rather than relying solely on manual judgment, I aim to build a data-driven prioritization system that helps content teams identify pages with the greatest opportunity for improvement. This project is designed to support better business decisions by providing evidence-based recommendations while keeping humans responsible for the final decision.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

### Decision
Which webpages should be prioritized for content review and refresh?

### Action
The SEO or content team will review and update the pages ranked as the highest priority before spending time on lower-priority pages.

### Cost of a Wrong Recommendation
If the recommendation is incorrect, the team may waste time improving pages with little business impact while overlooking pages that could generate greater organic traffic or user engagement.

### Why Data / Machine Learning?
Page performance depends on many factors such as impressions, clicks, CTR, engagement, and content freshness. Machine learning can combine these signals to provide a more reliable prioritization than simple rule-based methods.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import pandas as pd
import numpy as np

df = pd.read_csv("/content_refresh_anonymized.csv")

In [5]:
print(f"Number of pages: {len(df):,}")
print(f"Number of features: {df.shape[1]}")
print(f"Average impressions (90d): {df['impressions_90d'].mean():.2f}")
print(f"Average CTR: {df['ctr'].mean():.4f}")
print("\nTrend distribution:")
print(df["trend_direction"].value_counts())

Number of pages: 30,000
Number of features: 44
Average impressions (90d): 5200.37
Average CTR: 0.5107

Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Observations

The starter dataset contains **30,000 webpages** with **44 features**, providing sufficient information to analyze content performance from multiple perspectives. On average, each page received **5,200.37 impressions** over the past 90 days, indicating that many pages have measurable search visibility.

The trend distribution also shows that **16,262 pages are classified as "down"**, making it the largest category in the dataset. This suggests that a substantial number of pages may require review, while limited resources make it impractical to inspect every page manually.

These observations support my choice of the **Refresh / Content Opportunity Scoring** lane because a data-driven prioritization system can help content teams focus on the pages that are most likely to benefit from review and updates.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project is intended as a decision-support system rather than a decision-making system. The model can identify pages that appear to deserve further review based on observed patterns in the available data. It cannot prove that refreshing a page will cause higher rankings or increased traffic, nor can it predict Google's ranking algorithm. Final decisions should always involve human judgment.

## Self-check

- [x] Every section above is filled with both markdown explanations and supporting code.
- [x] The notebook runs from top to bottom without errors.
- [x] No client names, URLs, or private queries are included.
- [x] My claims use careful language such as "observed", "measured", "directional", and "decision-support".
- [x] The notebook is committed to my repository under `work/notebooks/`.